plugin/vsnip.vim

```diff
@@ -138,7 +138,7 @@ function! s:vsnip_set_text(type, copy) abort
     return
   endif
   execute 'normal! '.select.a:copy
-  call vsnip#selected_text(substitute(@v, '\n$', '', ''))
+  call vsnip#selected_text(@v)
```

Comment:
Use actual text for the selected text.
The formatting was moved to the `variable.vim`.